src/mz/src/bin/mz/main.rs

```diff
@@ -366,18 +365,22 @@ async fn main() -> Result<()> {
                         .max_duration(timeout)
                         .clamp_backoff(Duration::from_secs(1))
                         .retry_async(|_| async {
-                            let environment =
-                                get_region_environment(&client, &valid_profile, &region)
-                                    .await
-                                    .with_context(|| "Retrieving environment data.")?;
-                            if !environment.resolvable {
-                                bail!(String::from("Timeout expired enabling region."));
-                            }
+                            let region = get_region(&client, &cloud_provider, &valid_profile)
+                                .await
+                                .with_context(|| "Retrieving region data.")?;
 
-                            loop {
-                                if check_environment_health(&valid_profile, &environment)? {
-                                    break Ok(environment);
+                            if let Some(region_info) = region.region_info {
+                                if !region_info.resolvable {
+                                    bail!(String::from("Timeout expired enabling region."));
                                 }
+
+                                loop {
+                                    if check_region_health(&valid_profile, &region_info)? {
+                                        break Ok(region_info);
+                                    }
+                                }
+                            } else {
+                                bail!(String::from("Timeout expired enabling region."));
```

Comment:
just to confirm, this doesn't bail immediately? (It allows `retry_async` to continue retrying this function until the timeout?)

Reply:
Correct. It doesn't bail the whole process. It allows to keep retrying. After the timeout if it throws another bail it will exit the process with the bail error.